cpp/src/arrow/compute/light_array.h

```diff
@@ -348,11 +349,13 @@ class ARROW_EXPORT ExecBatchBuilder {
   ExecBatch Flush();
 
   int num_rows() const { return values_.empty() ? 0 : values_[0].num_rows(); }
+  bool is_full() const { return num_rows() == num_rows_max(); }
 
   static int num_rows_max() { return 1 << kLogNumRows; }
 
  private:
   static constexpr int kLogNumRows = 15;
+  static constexpr int64_t kAlignment = 512;
```

Comment:
`kAlignment` is inside of `ExecBatchBuilder`, it will always align its buffers to `512` bytes. `alignment_` is a member of ResizableArrayData so that you can specify different alignments for `ResizableArrayData`.